### ISSUE TITLE
End User is notified when Bürokratt is not working

### ISSUE THREAD

**Issue by rasmusei** [buerokratt/Buerokratt-Chatbot#43]
AS AN End User
I WANT TO have a clear understanding if Bürokratt is not working
SO THAT I could make contact using other channels

### Acceptance Criteria

**Business requirements**

- [x] When Bürokratt is not working and the widget is still active on the organization's website then Chatbot displays a message that Bürokratt is unavailable "Bürokratt ei ole hetkel saadaval"
- [x] Another option is when Bürokratt is not working then the widget on the organization's website is automatically disabled and not visible to End Users

**GUI**

- [ ] GUI makes a request to an appropriate `/healthz` or similar endpoint
- [ ] Above endpoint is requested if widget gets not `HTTP 200` response from Ruuter
- [ ] Above endpoint is requested periodically after every `30 seconds` (period must be configurable)

**Endpoints**

Service `/healthz/chat`

- [ ] Content of `/healthz/chat` endpoint is provided by DSL developments
- [ ] Chat widget can be configured to ping `/healthz/chat` endpoint within a predefined timeframe
- [ ] Chat widget can be configured to only appear on the website if `/healthz/chat` endpoint responds with `HTTP 200 OK`
- [ ] Chat widget can be configured to be inactive and provide a custom error message if `/healthz/chat` endpoint responds with `HTTP 503 Service Unavailable`

**Comment by turnerrainer** [comment]
**Before there is an active chat session established**

- A REST endpoint for GUI to turn to to get Bürokratt's status;
- GUI responds based on that status.

**After there is an active chat session established**

- Status is received via active (currently only) SSE session;
- GUI responds based on that status.

**Comment by ceirowe** [comment]
@rasmusei a few questions regarding GUI:

1. Should health checking happen on each initialization or only after an error occured / not OK status code recieved?
2. Should health checking stop pinging, if the status is OK 200 and restart after an error occurs?

**Comment by joonasroosalung** [comment]
@turnerrainer Uses existing /healthz endpoint on ruuter?

**Comment by turnerrainer** [comment]
> @turnerrainer Uses existing /healthz endpoint on ruuter?

No, this one needs a lot more attention. Maybe needs an additional grooming?

Just pinging definitely Ruuter won't do.

**Comment by joonasroosalung** [comment]
Rasa bot should have a /health or /healthcheck endpoint, will double check today/tomorrow

**Comment by ceirowe** [comment]
@turnerrainer GUI needs a way to test changes against working Ruuter.

**Comment by joonasroosalung** [comment]
Ruuter components already covered with existing v1 conf:
* Ruuter itself
* TIM
* Dmapper
* Resql

The existing conf https://github.com/buerokratt/Buerokratt-Chatbot/blob/main/DSL.Ruuter-v1.private/cs-get-components-healthz-status.json, which internally uses https://github.com/buerokratt/Buerokratt-Chatbot/blob/main/DSL.Ruuter-v1.private/cs-get-resql-healthz.json to check resql through an extra step. Don't remember the precise technical reason for that, likely due to http status response codes, Ruuter wants to return 200 almost always.

The existing endpoint responds with 
```
{"data":{"set_healthz_response":{"data":[{"name":"TIM","version":"1.0"},{"name":"RUUTER","version":"1.0"},{"name":"DMAPPER","version":"1.0"},{"name":"RESQL","version":"1.0"}]}},"error":null}
```
Version will be an empty string "" if any component is down and of course ther will be no response from the endpoint if ruuter itself is down.

**Comment by turnerrainer** [comment]
> > @turnerrainer Uses existing /healthz endpoint on ruuter?
> 
> No, this one needs a lot more attention. Maybe needs an additional grooming?
> 
> Just pinging definitely Ruuter won't do.

@joonasroosalung I'll take it back - the current `/healthz` endpoint is sufficient _in case_ it's also available for public Ruuter as well.

**Comment by sergeirudz** [comment]
PR: https://github.com/buerokratt/Chat-Widget/pull/20

**Comment by janinakimtrohlev** [comment]
BUG - right now if Bürokratt is not available it asks kontakt information. But if I as End User put my contact information, it tells that it is saved. But in History it says that contact information is given, but there is no contact information. 


![Image](https://github.com/buerokratt/Buerokratt-Chatbot/assets/122867977/28e76bee-1d84-428a-a4b8-8133419f1a54)

So it should be if I as end user put my contact information, then as CSA it will be shown in history.
